routes/authors.js

```diff
@@ -0,0 +1,109 @@
+
+var Hapi = require('hapi');
+
+exports.register = function(server, options, next){
+
+  server.route([
+  
+    //1. GET /authors 
+    {
+      method: 'GET',
+      path: '/authors',
+      handler: function(request, reply) {
+        var db = request.server.plugins['hapi-mongodb'].db;
+
+        db.collection('authors').find().toArray(function(err, authors){
+          if (err) {return reply('Internal MongoDB error', err);}
+          reply(authors);
+        });
+      }
+    },
+
+     // 2. GET /authors/{id}
+    {
+      method: 'GET',
+      path: '/authors/{id}',
+      handler: function(request, reply) {
+        var db       = request.server.plugins['hapi-mongodb'].db;
+        var ObjectID = request.server.plugins['hapi-mongodb'].ObjectID;
+        var author_id = ObjectID(encodeURIComponent(request.params.id));
+
+        db.collection('authors').findOne({"_id": author_id}, function(err, authors){
+          if (err) {
+            return reply('Internal MongoDB error',err);
+          }
+          reply(authors);
+        })
+      }
+    },
+
+    //3. POST /authors
+    {
+      method: 'POST',
+      path: '/authors',
+      handler: function(request, reply){
+        var db = request.server.plugins['hapi-mongodb'].db;
+      
+        var author = { message: request.payload.author.message };
```

Comment:
when you add an author, you are only adding a `message`? what is `author.message`? I would thought authors have a name and DOB and all that stuff no?